frontend/src/services/apiService.ts

```diff
@@ -1,18 +1,29 @@
+import axios from 'axios';
+import { environment } from '../utils/environment';
+
 const headers = (token?: string) => ({
   Authorization: `Bearer ${token}`,
   'Content-Type': 'application/json',
 });
 
-const apiRequest = (method: string) => async (
-  request: RequestInfo,
+const axiosInstance = axios.create({
+  baseURL: environment.baseUrl,
+  timeout: 1000,
+});
+
+const apiRequest = (method: 'GET' | 'PUT' | 'POST' | 'DELETE') => async (
+  url: string,
   token?: string,
```

Comment:
is it possible we can set the token via interceptor?

Reply:
Not sure how straightforward it will be with current auth0 implementation

Reply:
I think useAuth0 has to be wrapped in a react component at the moment, I'll see if there is a workaround.

Reply:
This seems like a fairly big task and more than one way of going about this so I opened #185